### ISSUE TITLE
Allow exceptions to feedback for certain exercises

### ISSUE THREAD

**Issue by mhelmetag** [exercism/rikki-ruby-analyzer#5]
From @cleeverz,

> Any time you submit an exercise on the ruby track where a conditional is used while iterating through a collection, you get this feedback from rikki:

> "Whenever you are looping through a collection and find yourself writing a conditional (if or unless) nested inside of the loop, take a moment to look through the available enumerable methods.
There are some very handy ones that might let you simplify."

> However, there are some exercises in which you are meant to implement your own, low level solution to a problem that one of these existing methods already has solved. In these cases this feedback is inapplicable at best, or could prove very confusing at worst, given the experience of a particular submitter.

> I suggest making a change to rikki that would repress this feedback in these cases.

The only blocking issue here is that rikki doesn't give exercise information to the API services. This shouldn't be that hard of a change. I'll reference a ticket for that when I write that up over in [exercism/rikki](https://github.com/exercism/rikki).

**Comment by kotp** [comment]
Good observation indeed.

**Comment by kytrinyx** [comment]
Yeah, this is a good point.

This tickled a memory for me. @kotp brought this up in exercism/rikki in https://github.com/exercism/rikki/issues/27

At that point I suggested that the analyzer is the right place for it, but I think I changed my mind. The analyzer should be completely agnostic. Rikki should be able to remove certain rules for certain exercises.

I'm going to go ahead and close this one, and add the observations to the open issue on the rikki repository.
